lib/scenic/adapters/postgres.rb

```diff
@@ -153,11 +155,49 @@ def update_materialized_view(name, sql_definition)
         raise_unless_materialized_views_supported
 
         IndexReapplication.new(connection: connection).on(name) do
-          drop_materialized_view(name)
-          create_materialized_view(name, sql_definition)
+          replace_materialized_view(name, sql_definition)
         end
       end
 
+      # Replaces a materialized view in the database.
+      #
+      # Recreate the materialized view under a temporary name and apply new
+      # indexes (if any). Drop previous materialized view, renames the
+      # temporary materialized view to its original name.
+      #
+      # This is typically called in a migration via {Statements#replace_view}.
+      #
+      # @param name The name of the view to update
+      # @param sql_definition The SQL schema for the updated view.
+      #
+      # @raise [MaterializedViewsNotSupportedError] if the version of Postgres
+      #   in use does not support materialized views.
+      #
+      # @return [void]
+      def replace_materialized_view(name, sql_definition)
+        raise_unless_materialized_views_supported
+
+        temp_mv_name = "#{name}_temp"
```

Comment:
This extension of the name might not be valid depending on the original length of the name. Perhaps we should use a static name that we know to be short. `_scenic_mv` or something?